lib/stats/stats-query.c

```diff
@@ -60,22 +59,17 @@ _add_counter_to_index(StatsCluster *sc, gint type)
       counter->name = counter_full_name;
     }
 
-  g_hash_table_insert(counter_index, counter_full_name, counter);
   sc->indexed_mask |= (1 << type);
```

Comment:
I don't think indexed_mask is used for anything now that the index hash is gone.

Reply:
the index probably caches the counter->name creation which is then used during query. 

But that could be formatted during the creation of the counter if we need it to be cached.

 It seems that right now the key is stored as a part of the hash table key and it is freed when the hash table entry is removed. We might as well push the allocation responsibility to the StatsCounterItem (although I don't like this memory overhead, we might have 100k counters, a name for each of those is not a trivial amount of RAM). But at the moment we do this allocation anyway.

Reply:
removed the index mantaining, and moved the name manipuation into the stats-registry that works directly on the `StatsCounterItem`s